src/engine/database/neo4j.rs

```diff
@@ -42,6 +47,33 @@ pub struct Neo4jEndpoint {
 }
 
 impl Neo4jEndpoint {
+    /// Returns a new [`Neo4jEndpoint`] from the provided values.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// # use warpgrapher::Error;
+    /// # use warpgrapher::engine::database::neo4j::Neo4jEndpoint;
+    /// #
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     let ne = Neo4jEndpoint::new(
+    ///         "127.0.0.1".to_string(),
+    ///         7687,
+    ///         "neo4j".to_string(),
+    ///         "password".to_string()
+    ///     );
+    /// #    Ok(())
+    /// # }
+    /// ```
+    pub fn new(host: String, port: u16, user: String, pass: String) -> Self {
```

Comment:
Stylistic question... in most cases, we've made these kinds of string configuration parameters &str types, so that it's possible to pass in strings like "127.0.0.1" without the caller having to call .to_string() on a number of arguments.  The downside is, of course, the overhead of an extra string copy, if someone would've actually passed in a real `String`. But they're typically short values, not many-kilobyte text files, so it's not that much overhead.... especially for something that's not in a critical loop -- one typically only sets up a database endpoint once.  Any particular reason you made the arguments `String` instead of `&str`?

(I know some people go all the way to parameterizing these kinds of arguments as Into<String>, that I've not been a fan of that, as it clutters up the type signature of the function call too much for my liking.)

Reply:
I was operating under the assumption that the arguments for this function in particular will be derived from some source in real time (config file) and will likely be strings. Providing something like a password as a `&str` I suspect with would be the exception rather than the norm, and in those few cases calling `.to_string()` would not be a major inconvenience. 

I'm open to alternatives, however. Would you be in favor on switching the inputs to `&str` and then calling `.to_string()` on each of them in the constructor (since the fields in the struct are `String`s. ?

Reply:
Good point.  I'm good with `String`s.